win32ss/user/ntuser/ntstubs.c

```diff
@@ -346,16 +346,57 @@ NtUserSetSysColors(
    return Ret;
 }
 
-DWORD
+BOOL APIENTRY UserUpdateInputContext(PIMC pIMC, DWORD dwType, DWORD_PTR dwValue)
+{
+    PTHREADINFO pti = GetW32ThreadInfo();
+    PTHREADINFO ptiIMC = UserHMGetHandle(pIMC);
+
+    if (pti->ppi != ptiIMC->ppi)
+        return FALSE;
+
+    switch (dwType)
+    {
+        case UIC_CLIENTIMCDATA:
+            if (pIMC->dwClientImcData)
+                return FALSE;
+            pIMC->dwClientImcData = dwValue;
+            break;
+
+        case UIC_IMEWINDOW:
+            pIMC->hImeWnd = (HWND)dwValue;
```

Comment:
Shouldn't there be some sort of validation that the given value is indeed a valid window handle?

Reply:
Validation added.